llm_bench/python/convert.py

```diff
@@ -346,33 +345,99 @@ def convert_causal_lm(args):
         del pt_model
         gc.collect()
 
-    model = OVModelForCausalLM.from_pretrained(
-        args.model_id,
-        export=True,
-        compile=False,
-        trust_remote_code=True,
-        load_in_8bit=False,
-        config=AutoConfig.from_pretrained(args.model_id, trust_remote_code=True),
-    )
-    end = time.perf_counter()
+    if args.bettertransformer:
+        model = AutoModelForCausalLM.from_pretrained(
+            args.model_id,
+            trust_remote_code=True,
+            config=AutoConfig.from_pretrained(args.model_id, trust_remote_code=True)
+        )
 
-    log.info(f'Conversion total time {end - start}s')
-    if args.precision == 'FP16':
-        model.half()
-    ov_out_dir = Path(args.output_dir) / 'pytorch/dldt' / precision
-    save_tokenizer(tok, ov_out_dir)
+        model = model.to_bettertransformer()
 
-    start1 = time.perf_counter()
-    model.save_pretrained(ov_out_dir)
-    end1 = time.perf_counter()
-    log.info(f'Serialization total time {end1 - start1}s')
+        onnx_config_constructor = TasksManager.get_exporter_config_constructor(model=model, exporter="onnx", task="text-generation")
+        config = onnx_config_constructor(model.config, use_past=True, use_past_in_inputs=True)
+
+        # for better transformers we need sequence lenght to be not 1 to make a correct trace
+        # patch generate_dummy_inputs in the config
+
+        def pathed_generate_dummy_inputs(self, *args, **kwargs):
+            dummy_inputs = self._original_generate_dummy_inputs(*args, **kwargs)
+            if 'input_ids' in dummy_inputs and dummy_inputs['input_ids'].shape[1] == 1:
+                dummy_inputs['input_ids'] = torch.cat([dummy_inputs['input_ids'], dummy_inputs['input_ids']], dim=-1)
+                attention_mask = dummy_inputs['attention_mask']
+                dummy_inputs['attention_mask'] = torch.cat([attention_mask, attention_mask.new_ones((attention_mask.shape[0], 1))], dim=-1)
+            return dummy_inputs
+
+        # also as we are not using intel-optimimum model class for save/store the model we also need to patch inputs to
```

Comment:
Could you please check, I think it does not required anymore with new optimum version
https://github.com/huggingface/optimum/blob/main/optimum/exporters/onnx/config.py#L97